google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -255,24 +268,32 @@ private void notifyListeners(CloudSdkService sdkService, SdkStatus status) {
     UPDATE
   }
 
+  enum ManagedSdkJobResult {
```

Comment:
do we really need another status enum in addition to the Sdk status we already have? (the answer may be yes, just wondering if it can be simplified).

Reply:
I thought to combine them together, i.e. `INSTALL`, `UPDATE`, `UP_TO_DATE` in a single enum. However the problem is `FutureCallback` and future in general does not receive any result on failure. We wouldn't know if it was install or update since we use generic callback.

Reply:
interesting. I need to explore the code more then to see if I have anything to add here.

Reply:
is all this just for showing a certain message in the UI (`ManagedCloudSdkUiPresenter#notifyManagedSdkJobSuccess`)?

Reply:
for this, and also to handle update slightly differently in case of a failure, keeping the rest of the logic the same.

Reply:
can you point me to that code (the other case of handling update differently during failure) please?

Reply:
nm. I found it. ManagedCloudSdkService#onFailure.

Reply:
actually that is using `ManagedSdkJobType`. the only place I could find actually using `ManagedSdkJobResult` is the Ui presenter for showing notifications. maybe I'm missing something?

Reply:
ah yes, you mean `ManagedSdkJobResult` - this one only used to check if notification is needed, correct.